### PR TITLE
adding post vector search ranking

### PR DESCRIPTION
**Pull request by fshabashev**
## Description

adding post vector search reranking and filtering in order to increase accuracy of the vector search

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Location: Specify the URL or path for testing.
 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

path to the additional unit test:
tests/integrations/utilities/rag/rerankers/test_openai_reranker.py

steps to test end-to-end:
1. start mindsdb locally
2. run "using agents with retrieval" from  mindsdb_python_sdk
3. check that output of the script is correct

### CODE REVIEW

**Comment by dusvyat** [comment]
looks good to me.

In a follow up PR we should add support for re-ranking with Nvidia and hugging face

**Review by dusvyat — APPROVED** [review]

File: requirements/requirements.txt [hunk 1]
```diff
@@ -47,3 +47,4 @@ langfuse==2.35.0
 lark
 prometheus-client==0.20.0
 transformers >= 4.42.4
+openai
```
Comment (dusvyat) [inline requirements/requirements.txt:50]: this is causing some of the CI/CD to fail, we should remove it

**Comment by fshabashev** [comment]
Screenshot from 2024-10-07 18-51-23
Screenshot from 2024-10-07 18-49-27

screenshots of the completed tests

**Comment by dusvyat** [comment]
nice, looks good to go

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 2]
```diff
@@ -29,6 +29,8 @@
 
 DEFAULT_EMBEDDINGS_MODEL = "BAAI/bge-base-en"
 
+DEFAULT_RERANKING_MODEL = "gpt-4o"
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:32]: let's remove this change

Comment (fshabashev) [reply mindsdb/integrations/handlers/rag_handler/settings.py]: just removed it

**Pull request merged**